pkg/feature/flags.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+Package feature provides a global "Feature Flag" facility used to toggle entire features on or off.
+
+To "register" a new feature,
+1. add a new constant
+2. add that constant to the `allFeatures` slice
+
+Initialize the flags, enabling desired features:
+
+    feature.Flags().Enable(<feature-constant>)
+
+To then circuit-break functionality behind a feature flag.
+
+    if feature.Flags().IsEnabled(<feature-constant>) {
+        ...
+    }
+*/
+package feature
+
+import "fmt"
+
+// Names of features that can be toggled
+const (
+	Noop        = "noop"
```

Comment:
What's the intention behind `noop`?

Reply:
Honestly?  It was born out of testing — needing to have a "feature" that could be verified enabled or disabled.

But then as I used it, thought it would be nice to differentiate between `YTTEXPERIMENTS=` containing an empty string, which currently panics....  (which as I wrote that sentence don't like that as an experience... 🤔 ).

...

Yeah, I'm gonna document this as "intended for testing" thing... _and_ **not** panic when `YTTEXPERIMENTS=` is an empty string.